atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/bucketingthings/SweepAssignedBucketStoreKeyPersisterTest.java

```diff
@@ -0,0 +1,272 @@
+                ShardAndStrategy.thorough(212),
+                ShardAndStrategy.nonSweepable());
+
+        List<Cell> cells = shardsAndStrategies
+                .map(SweepAssignedBucketStoreKeyPersister.INSTANCE::sweepBucketPointerCell)
+                .collect(Collectors.toList());
+
+        assertThat(cells).doesNotHaveDuplicates();
+    }
+
+    @ParameterizedTest
+    @MethodSource("goldenSweepBucketPointerCells")
+    public void shardsAndStrategiesMapToHistoricSweepBucketPointerCells(
+            ShardAndStrategy shardAndStrategy, Cell expectedCell) {
+        Cell actualCell = SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketPointerCell(shardAndStrategy);
+        assertThat(actualCell).isEqualTo(expectedCell);
+    }
+
+    @Test
+    public void differentBucketIdentifiersMapToDifferentSweepBucketRecordsCells() {
+        Set<Integer> bucketIdentifiers = Set.of(0, 1, 123123, Integer.MAX_VALUE);
+        List<Cell> cells = bucketIdentifiers.stream()
+                .map(SweepAssignedBucketStoreKeyPersister.INSTANCE::sweepBucketRecordsCell)
+                .collect(Collectors.toList());
+        assertThat(cells).doesNotHaveDuplicates();
+    }
+
+    @ParameterizedTest
+    @MethodSource("goldenSweepBucketRecordsCells")
+    public void bucketIdentifiersMapToHistoricSweepBucketRecordsCells(long bucketIdentifier, Cell expectedCell) {
+        Cell actualCell = SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketRecordsCell(bucketIdentifier);
+        assertThat(actualCell).isEqualTo(expectedCell);
+    }
+
+    // TODO(mdaudali): Should we add validation to ensure that bucket identifier for bucket and passed in above >= 0?
+
+    @Test
+    // A more rigorous test would likely involve some generator for the various inputs into each cell
+    // Perhaps an interesting place to leverage property-based testing
+    public void cellsDoNotOverlap() {
+        List<Cell> cells = Stream.concat(
+                        Stream.concat(
+                                Stream.of(
+                                        SweepAssignedBucketStoreKeyPersister.INSTANCE
+                                                .sweepBucketAssignerStateMachineCell(),
+                                        SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketRecordsCell(0)),
+                                buckets().stream()
+                                        .map(SweepAssignedBucketStoreKeyPersister.INSTANCE::sweepBucketsCell)),
+                        Stream.of(SweeperStrategy.values())
+                                .map(v -> SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketPointerCell(
+                                        ShardAndStrategy.of(0, v))))
+                .collect(Collectors.toList());
+        assertThat(cells).doesNotHaveDuplicates();
+    }
+
+    private static Set<Bucket> buckets() {
+        Set<Integer> shardsAndBucketIdentifiers = Set.of(0, 1, 123123, Integer.MAX_VALUE);
```

Comment:
Let me know if I've gone too far with the cartesian products!

(And sure, a shard isn't supposed to go above 256, but it's a minor liberty here...)